pkg/specgen/generate/kube/kube.go

```diff
@@ -381,6 +381,22 @@ func ToSpecGen(ctx context.Context, opts *CtrSpecGenOptions) (*specgen.SpecGener
 				Options: options,
 			}
 			s.Volumes = append(s.Volumes, &cmVolume)
+		case KubeVolumeTypeCharDevice:
+			// We are setting the path as hostPath:mountPath to comply with DeviceFromPath (https://github.com/containers/podman/blob/eb26fa45f1326191dea27f2afabf82cb8b934140/pkg/specgen/generate/config_linux.go#L72)
```

Comment:
The link will get out of date very quickly, so I prefer not to use one. Stating that it complies with `pkg/specgen/generate.DeviceFromPath` may be more robust.

Same for the comment below.

Reply:
@jwhonce PTAL at this - I believe you have a PR trying to solve this, when it looks already solved?